test/new-e2e/tests/language-detection/node_test.go

```diff
@@ -13,15 +13,25 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// nodeMajor is the major version of nodejs that will be installed
+const nodeMajor = 20
+
 //go:embed etc/node_server.js
 var nodeProg string
 
 func (s *languageDetectionSuite) installNode() {
-	s.Env().VM.Execute(
-		"curl -fsSL https://deb.nodesource.com/setup_20.x | sudo -E bash - && " +
-			"sudo apt-get install -y nodejs")
+	// Installation instructions taken from https://github.com/nodesource/distributions
+	s.Env().VM.Execute("sudo apt-get update")
+	s.Env().VM.Execute("sudo apt-get install -y ca-certificates curl gnupg")
+	s.Env().VM.Execute("sudo mkdir -p /etc/apt/keyrings")
+	s.Env().VM.Execute("curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg")
+	s.Env().VM.Execute(fmt.Sprintf("echo \"deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_%d.x nodistro main\" | sudo tee /etc/apt/sources.list.d/nodesource.list", nodeMajor))
+	s.Env().VM.Execute("sudo apt-get update")
+	s.Env().VM.Execute("sudo apt-get install nodejs -y")
```

Comment:
How is this set up - is this reusing the shell?

Reply:
Indeed, golang ssh client allows one call per session

> A Session only accepts one call to Run, Start, Shell, Output, or CombinedOutput.

Source: https://pkg.go.dev/golang.org/x/crypto/ssh#Session.Run

we should add instructions on this. For this case you could join instructions and run them as a script. If you are interested in each single command result, then better use one execute per command